parlai/core/metrics.py

```diff
@@ -300,92 +377,30 @@ def normalize_answer(s):
     return s
 
 
-def aggregate_task_reports(reports, tasks, micro=False):
+def aggregate_named_reports(named_reports: Dict[str, Dict[str, Metric]]):
     """
-    Aggregate separate task reports into a single report.
-
-    :param reports: list of report dicts from separate tasks
-    :param tasks: list of tasks
-    :param micro: average per example if True, else average over t
+    Aggregate metrics from multiple reports.
 
-    :return: aggregated report dicts
+    :param reports: Dict of tasks -> metrics.
     """
-    if len(reports) == 1:
-        # singular task
-        return reports[0]
-    # multiple tasks, aggregate metrics
-    metrics = {}
-    exs = {}
-    total_report = {'tasks': {}}
-    # collect metrics from all reports
-    for i, report in enumerate(reports):
-        total_report['tasks'][tasks[i]] = report
-        for metric, val in report.items():
-            if metric == 'exs':
-                exs[tasks[i]] = val
-            else:
-                metrics.setdefault(metric, {})[tasks[i]] = val
-    # now aggregate
-    total_exs = sum(exs.values())
-    total_report['exs'] = total_exs
-    for metric, task_vals in metrics.items():
-        if all([isinstance(v, Number) for v in task_vals.values()]):
-            if micro:
-                # average over the number of examples
-                vals = [task_vals[task] * exs[task] for task in tasks]
-                total_report[metric] = round_sigfigs(sum(vals) / total_exs, 4)
-            else:  # macro
-                # average over tasks
-                vals = task_vals.values()
-                total_report[metric] = round_sigfigs(sum(vals) / len(vals), 4)
-    # add a warning describing how metrics were averaged across tasks.
-    total_report['warning'] = 'metrics are averaged across tasks'
-    if micro:
-        total_report['warning'] += ' and weighted by the number of examples ' 'per task'
-    return total_report
-
-
-def aggregate_metrics(reporters):
+    # reporters is a list of teachers or worlds
+    m: Dict[str, Metric] = {}
+    for task_id, task_report in named_reports.items():
+        for each_metric, value in task_report.items():
+            m[each_metric] = m.get(each_metric, None) + value
+            if len(named_reports) > 1:
+                m[f'{task_id}/{each_metric}'] = value
```

Comment:
So the choice of a slash notation in the key string here confuses me a bit. Why not have a second dictionary within the first?